util.js

```diff
@@ -0,0 +1,92 @@
+const IsCanonicalBase64 = require('is-canonical-base64')
+const { isFeedType, isMsgType, isBlobType } = require('ssb-ref')
+
+module.exports = {
+  decorateBFE,
+  findClassicTypeFormat,
+  sortByNames,
+}
+
+const encryptedTypeRegex = IsCanonicalBase64('', '\\.box\\d*')
+const sigTypeRegex = IsCanonicalBase64('', '\\.sig\\.[a-zA-Z0-9]+')
+
+const isEncryptedType = (input) => encryptedTypeRegex.test(input)
+const isSigType = (input) => sigTypeRegex.test(input)
+
+function decorateBFE(types) {
+  return types.map((type) => {
+    return {
+      ...type,
+      code: Buffer.from([type.code]),
+      formats: type.formats.map((format) => {
+        return {
+          ...format,
+          code: Buffer.from([format.code]),
+          // TFCode: Buffer.from([type.code, format.code]),
+          sigilSuffixRegexp: buildSigilSuffixRegexp(type, format),
+        }
+      }),
+    }
+  })
+}
+
+function buildSigilSuffixRegexp(type, format) {
+  return type.sigil || format.suffix
+    ? IsCanonicalBase64(
+        type.sigil || '',
+        (format.suffix && format.suffix.replace('.', '\\.')) || '',
+        format.key_length
+      )
+    : // NOTE this assumes all sigil / suffic encodings are base64
+      undefined
+}
+
+function sortByNames(types) {
```

Comment:
ohh, this doesn't sort by name :) It converts it into a dict ;)

Reply:
good spotting!

Reply:
renamed `definitionsToDict`